server/server.go

```diff
@@ -31,25 +31,50 @@ func New(runtimePath, sandboxDir, containerDir string) (*Server, error) {
 		return nil, err
 	}
 	sandboxes := make(map[string]*sandbox)
+	containers := make(map[string]*container)
 	return &Server{
 		runtime:    r,
 		sandboxDir: sandboxDir,
 		state: &serverState{
-			sandboxes: sandboxes,
+			sandboxes:  sandboxes,
+			containers: containers,
 		},
 	}, nil
 }
 
 type serverState struct {
-	sandboxes map[string]*sandbox
+	sandboxes  map[string]*sandbox
+	containers map[string]*container
 }
 
 type sandbox struct {
-	name   string
-	logDir string
-	labels map[string]string
+	name       string
+	logDir     string
+	labels     map[string]string
+	containers []*container
 }
 
 func (s *Server) addSandbox(sb *sandbox) {
 	s.state.sandboxes[sb.name] = sb
 }
+
+func (s *Server) hasSandbox(name string) bool {
+	_, ok := s.state.sandboxes[name]
+	return ok
+}
+
+func (s *sandbox) addContainer(c *container) {
+	s.containers = append(s.containers, c)
+}
+
+type container struct {
+	name    string
+	logPath string
+	labels  map[string]string
+	sandbox *sandbox
```

Comment:
I think we can still move it to oci with a string for sandbox if we need a back reference.